lib/queuePopulator/LogReader.js

```diff
@@ -318,33 +320,47 @@ class LogReader {
             }
             return key;
         }
-        // TODO: handle this better, this is too specific to notifications
-        // temp fix
-        const notifExtension = this._extensions.find(ext => (
-            ext instanceof NotificationQueuePopulator
-        ));
-        if (notifExtension) {
-            notifExtension.filter({
-                type: entry.type,
-                bucket: record.db,
-                key: _transformKey(entry.key),
-                value: entry.value || '{}',
-            });
-        } else {
-            if (entry.key === undefined || entry.value === undefined) {
-                return;
+
+        function _executeFilter(ext, entry, cb) {
+            if (typeof ext.filterAsync === 'function') {
+                ext.filterAsync(entry, cb);
+            } else {
+                ext.filter(entry);
+                cb();
+            }
+        }
+
+        async.eachSeries(this._extensions, (ext, next) => {
+            let entryValue = entry.value;
+            const notifExtension = ext instanceof NotificationQueuePopulator;
+            if (notifExtension) {
+                entryValue = entry.value || '{}';
             }
-            this._extensions.forEach(ext => ext.filter({
+            if (!notifExtension &&
```

Comment:
Why can't this logic be in the notification extension itself?

Reply:
If don't handle this here, the entries with no value, typically `del` operation on an object (unversioned) will be filtered out here.

Reply:
ah I see. In order not to have to change other existing extension filters, we can do this indeed, but maybe it should be reworked at some point (we only have a couple extensions to modify).